provider/queue/queue.go

```diff
@@ -124,7 +80,26 @@ func (q *Queue) work() {
 
 		for {
 			if c == cid.Undef {
-				k, c = q.nextEntry()
+				head, e := q.getQueueHead()
+
+				if e != nil {
+					log.Errorf("error querying for head of queue: %s, stopping provider", e)
+					return
```

Comment:
If we get an error querying for head, then the provider will not work so I stop it. Bad idea?

Reply:
The other option is to keep attempting the query while respecting some backoff mechanism. Is that worth the extra effort?

Reply:
Let's leave it like this for now. We _shouldn't_ get an error.